File: app/src/main/java/com/verbatoria/di/login/LoginComponent.java
package com.verbatoria.di.login;

import com.verbatoria.ui.login.view.login.LoginActivity;
import com.verbatoria.ui.login.view.recovery.RecoveryActivity;
import com.verbatoria.ui.login.view.sms.SMSConfirmationActivity;

import dagger.Subcomponent;

/**
 * Компонент Даггера для логина
 *
 * @author nikitaremnev
 */
@Subcomponent(modules = {LoginModule.class})
@LoginScope
public interface LoginComponent {

    void inject(LoginActivity loginActivity);

    void inject(RecoveryActivity recoveryActivity);

    void inject(SMSConfirmationActivity smsConfirmationActivity);

}

File: app/src/main/java/com/verbatoria/di/login/LoginComponent.kt
package com.verbatoria.di.login

import com.verbatoria.di.base.BaseInjector
import com.verbatoria.ui.login.LoginActivity
import dagger.Subcomponent

/**
 * @author n.remnev
 */

@Subcomponent(modules = [LoginModule::class])
@LoginScope
interface LoginComponent : BaseInjector<LoginActivity> {

    @Subcomponent.Builder
    interface Builder {

        fun build(): LoginComponent

    }

}
